packages/metal-dom/src/domNamed.js

```diff
@@ -521,6 +521,34 @@ export function parent(element, selector) {
 	return closest(element.parentNode, selector);
 }
 
+/**
+ * Inserts a node before first child of the parent. If child is a HTML string
+ * it will be converted to document fragment before prepending it to the parent.
+ * @param {!Element} parent The node to prepend to.
+ * @param {!(Element|NodeList|string)} child The thing who must be prepended
+ * @return {!Element} The prepended child
+ */
+export function prepend(parent, child) {
+	if (isString(child)) {
+		child = buildFragment(child);
+	}
+
+	if (!isNodeListLike(child) && !isDefAndNotNull(parent.firstChild)) {
+		parent.appendChild(child);
+	}
+
+	if (isNodeListLike(child)) {
+		const childArr = Array.prototype.slice.call(child);
+		for (let i = 0; i < childArr.length; i++) {
+			parent.appendChild(childArr[i]);
+		}
+	} else {
+		parent.insertBefore(child, parent.firstChild);
```

Comment:
When calling this on a parent that has no children, this block will run even though line 537 has already added the child.

Reply:
Hey @Robert-Frampton , what do you think that function return parent? User probably can get info about childNodes easily, instead actually returning child. I've sent a commit that fixes the problem on running though and returns parent.

Reply:
Hey @diegonvs, we should probably keep it the same as the `dom.append` method which returns the child.